### ISSUE TITLE
Mingw-m64 build failing on Appveyor

### ISSUE THREAD

**Issue by wbhart** [wbhart/flint2#439]
I'm having trouble figuring out why the 64 bit Appveyor build for Mingw-w64 is failing.

The MPFR version changed to mpfr-4.0.0 and I've successfully got it to build mpfr-4.0.0. But now Flint says the MPFR directory is wrong. I don't see why it says that. The .appveyor_build_dependencies.sh file is surely where that is selected, and it definitely uses mpfr-4.0.0 which should now exist.

Do you have any idea about this @isuruf ?

**Comment by wbhart** [comment]
https://ci.appveyor.com/project/wbhart/flint2/build/1.0.388/job/q3s128t9fiw7yq2o

**Comment by isuruf** [comment]
Actually the mpfr build fails, but the output is redirected to /dev/null. MPFR build fails with 
```
In file included from mpfr-impl.h:99:0,
                 from add_ui.c:24:
add_ui.c: In function `mpfr_add_ui':
mpfr-gmp.h:86:33: error: `mpfr_clz_tab' undeclared (first use in this function)
 #define __clz_tab               mpfr_clz_tab
                                 ^
/c/projects/flint2/mpir-2.7.2/longlong.h:600:39: note: in expansion of macro `__
clz_tab'
     (count) = W_TYPE_SIZE + 1 - __a - __clz_tab[__xr >> __a];  \
                                       ^~~~~~~~~
add_ui.c:69:5: note: in expansion of macro `count_leading_zeros'
     count_leading_zeros (cnt, (mp_limb_t) u);
     ^~~~~~~~~~~~~~~~~~~
mpfr-gmp.h:86:33: note: each undeclared identifier is reported only once for eac
h function it appears in
 #define __clz_tab               mpfr_clz_tab
                                 ^
/c/projects/flint2/mpir-2.7.2/longlong.h:600:39: note: in expansion of macro `__
clz_tab'
     (count) = W_TYPE_SIZE + 1 - __a - __clz_tab[__xr >> __a];  \
                                       ^~~~~~~~~
add_ui.c:69:5: note: in expansion of macro `count_leading_zeros'
     count_leading_zeros (cnt, (mp_limb_t) u);
     ^~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:800: add_ui.lo] Error 1
make[2]: Leaving directory '/c/projects/flint2/mpfr-4.0.0/src'
make[1]: *** [Makefile:469: all] Error 2
make[1]: Leaving directory '/c/projects/flint2/mpfr-4.0.0/src'
make: *** [Makefile:498: all-recursive] Error 1
```

**Comment by isuruf** [comment]
Looks like mpfr-4.0.0 and mpir-2.7.2 are not compatible

**Comment by wbhart** [comment]
And MPIR and Flint are incompatible on Windows. I knew the day would come...

**Comment by wbhart** [comment]
Since MPIR-3.0.0 and Flint are no longer compatible on Windows and MPIR-2.7.2 and MPFR-4.0.0 are no longer compatible, I think we have to restrict to using MPIR-2.7.2 and MPFR-3.6.1. We can obtain the latter from [1].

Eventually it's possible Flint won't be compatible with MPIR-2.7.2 or MPFR-3.6.1 in which case we will lose Windows support altogether. Switching to GMP is possible, but the Flint GMP compatibility wrapper for Windows does come with some performance issues.

[1] http://www.mpfr.org/mpfr-3.1.6/

**Comment by wbhart** [comment]
It seems to currently build with mpfr-4.0.0 and mpir-2.7.2, which is fine. We could also support mpir-3.0.0 on Windows, but that is not what this ticket is about.
